scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -339,6 +339,22 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    */
   def distinct: SCollection[T] = this.pApply(Distinct.create[T]())
 
+  /**
+   * Returns a new SCollection with distinct elements using given function to obtain a
+   * representative value for each input element.
+   *
+   * @param f The funciton to use to get representative values.
+   * @tparam U The type of representative values used to dedup.
+   * @group transform
+   */
+  def distinctWithRepresentativeValueFn[U](f: T => U)(implicit ctu: ClassTag[U])
```

Comment:
agree with @nevillelyh this can be called `distinctBy`

Reply:
Yes, we have a few extra functions as part of our codebase, where we have `distinctByKey`. I was confused if we should follow the Beam names, so I kept it as `distinctWithRepresentativeValueFn`, but a simple `distinctBy` makes more sense.
I will also add a `distinctByKey` for `PairSCollectionFunctions`.